frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/image/image_placeholder.dart

```diff
@@ -37,65 +39,116 @@ class ImagePlaceholderState extends State<ImagePlaceholder> {
-            controller.close();
-            WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
-              await insertAIImage(url);
-            });
-          },
-          onSelectedNetworkImage: (url) {
-            controller.close();
-            WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
-              await insertNetworkImage(url);
-            });
-          },
-        );
-      },
-      child: DecoratedBox(
-        decoration: BoxDecoration(
-          color: Theme.of(context).colorScheme.surfaceVariant,
+      child: FlowyHover(
+        style: HoverStyle(
           borderRadius: BorderRadius.circular(4),
         ),
-        child: FlowyHover(
-          style: HoverStyle(
-            borderRadius: BorderRadius.circular(4),
-          ),
-          child: SizedBox(
-            height: 48,
-            child: Row(
-              children: [
-                const HSpace(10),
-                const FlowySvg(
-                  FlowySvgs.image_placeholder_s,
-                  size: Size.square(24),
-                ),
-                const HSpace(10),
-                FlowyText(
-                  LocaleKeys.document_plugins_image_addAnImage.tr(),
-                ),
-              ],
-            ),
+        child: SizedBox(
+          height: 52,
+          child: Row(
+            children: [
+              const HSpace(10),
+              const FlowySvg(
+                FlowySvgs.image_placeholder_s,
+                size: Size.square(24),
+              ),
+              const HSpace(10),
+              FlowyText(
+                LocaleKeys.document_plugins_image_addAnImage.tr(),
+              ),
+            ],
           ),
         ),
       ),
     );
+    if (PlatformExtension.isDesktopOrWeb) {
+      child = AppFlowyPopover(
+        controller: controller,
+        direction: PopoverDirection.bottomWithCenterAligned,
+        constraints: const BoxConstraints(
+          maxWidth: 540,
+          maxHeight: 360,
+          minHeight: 80,
+        ),
+        clickHandler: PopoverClickHandler.gestureDetector,
+        popupBuilder: (context) {
+          return UploadImageMenu(
+            onSelectedLocalImage: (path) {
+              controller.close();
+              WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+                await insertLocalImage(path);
+              });
+            },
+            onSelectedAIImage: (url) {
+              controller.close();
+              WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+                await insertAIImage(url);
+              });
+            },
+            onSelectedNetworkImage: (url) {
+              controller.close();
+              WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+                await insertNetworkImage(url);
+              });
+            },
+          );
+        },
+        child: child,
+      );
+    } else {
```

Comment:
No need for else block if you return instead of assign child in the if-statement above